crates/api/src/fetch.rs

```diff
@@ -34,17 +42,29 @@ pub fn serialize_op_ids(value: Vec<OpId>) -> bytes::Bytes {
 /// Deserialize list of op ids.
 pub fn deserialize_op_ids(value: bytes::Bytes) -> K2Result<Vec<OpId>> {
     let op_ids = OpIds::decode(value).map_err(K2Error::other)?;
-    let vec = Vec::from(op_ids);
-    Ok(vec)
+    Ok(op_ids.into())
+}
+
+/// Serialize list of ops for sending over the wire.
+pub fn serialize_ops(value: Vec<MetaOp>) -> bytes::Bytes {
+    let ops = Ops::from(value).encode_to_vec();
+    bytes::Bytes::copy_from_slice(&ops)
 }
 
 /// Trait for implementing a fetch module to fetch ops from other agents.
 pub trait Fetch: 'static + Send + Sync + std::fmt::Debug {
-    /// Add op ids to be fetched.
-    fn add_ops(
+    /// Add op ids to be fetched from a peer.
+    fn request_ops(
         &self,
-        op_list: Vec<OpId>,
-        source: AgentId,
+        ops: Vec<OpId>,
+        from: AgentId,
+    ) -> BoxFut<'_, K2Result<()>>;
+
+    /// Send held ops as a response to a requesting peer.
+    fn handle_op_request(
```

Comment:
Nice, I like the naming now. It's clear these are a pair

Reply:
This should not be in the external api.

1. In fetch module constructor--register with transport module to receive module messages.
2. When we receive a request, send that request to the response queue.
3. The request will come with a URL, so you don't need to look up by agent id... just send it to where we got the request from.

Reply:
Of course, the concrete implementation can have this function be private that you can call for testing to bypass actually receiving things in the transport.

Reply:
`recv_module_msg` is sync, the response queue sender is async - use a tokio task for that?

Reply:
The pattern we've started to establish (but should evaluate if appropriate on a case-by-case basis) is to use a bounded channel, then [try_send](https://docs.rs/async-channel/2.3.1/async_channel/struct.Sender.html#method.try_send) and if the send fails, it means the system is overloaded, so log the problem and drop the request.

It seems like that is appropriate here.

Reply:
Great.